src/VisualStudio/Core/Def/Implementation/Options/RoamingVisualStudioProfileOptionPersister.cs

```diff
@@ -50,20 +58,33 @@ public RoamingVisualStudioProfileOptionPersister(IThreadingContext threadingCont
         {
             Contract.ThrowIfNull(globalOptionService);
 
-            _settingManager = (ISettingsManager)serviceProvider.GetService(typeof(SVsSettingsPersistenceManager));
             _globalOptionService = globalOptionService;
 
-            // While the settings persistence service should be available in all SKUs it is possible an ISO shell author has undefined the
-            // contributing package. In that case persistence of settings won't work (we don't bother with a backup solution for persistence
-            // as the scenario seems exceedingly unlikely), but we shouldn't crash the IDE.
-            if (_settingManager != null)
-            {
-                var settingsSubset = _settingManager.GetSubset("*");
-                settingsSubset.SettingChangedAsync += OnSettingChangedAsync;
-            }
+            var serviceContainer = serviceProvider.GetService<SVsBrokeredServiceContainer, IBrokeredServiceContainer>();
+            _serviceBroker = serviceContainer.GetFullAccessServiceBroker();
```

Comment:
Please call `Assumes.Present(serviceContainer);` before dereferencing the result of `GetService`. 
But ... you know what? Don't do any of this at all. You're in an MEF importing constructor, and as such this whole thing should be free-threaded. You shouldn't be calling `GetService` here at all. Can you move it to another method that isn't called in a MEF callstack?